KS-20-05-inverter

DenisGubernatorov opened the pull request:
1) Класс InverterRobot с методами say(), setModifier(), списком фраз и не инициализированным полем inverter, принимающим лямбду 2) Метод say() печатает фразу из лямбды если inverter не null или случайную фразу из списка фраз 3) setModifier устанавливает лямбду, которая берет случайную фразу из списка фраз и возвращает ее в перевернутом виде

DenisGubernatorov committed:
KS-20-05-inverter
1) Класс InverterRobot с методами say(), setModifier(), списком фраз и не инициализированным полем inverter, принимающим лямбду
2) Метод say() печатает фразу из лямбды если inverter не null или случайную фразу из списка фраз
3) setModifier устанавливает лямбду, которая берет случайную фразу из списка фраз и возвращает ее в перевернутом виде

ievetrovdev commented on src/main/kotlin/lesson20/lesson20_task5.kt:
```diff
@@ -1 +1,30 @@
 package lesson20
+
+class InverterRobot {
+    private var inverter: (() -> String)? = null
```
Лучше добавить свойство в класс с модификатором по умолчанию (String) -> String = { it }

DenisGubernatorov committed:
KS-20-05-inverter
1) Рефакторинг. inverter инициализируется лямбдой по умолчанию.
2) Изменение лямбды в setModifier. Что бы тоже соответствовала функциональному типу (String)->String

ievetrovdev reviewed (approved):
Все верно, отлично!
